### PR TITLE
Lowercase emails

### PR DESCRIPTION
**Pull request by aaronoppenheimer**
Added various bits so emails are properly lowercased:
- when creating a member object (using either the membermanager or when saving a member object)
- when inviting members
- added login authentication bit to compare to database without case sensitivity
- added a test case or two

Also
- fixed bug in email-address change process so confirmation goes to new address, not old

Plus
- added some helpful debugging things: ability to disable captcha from settings, printing out email body when email is disabled

### CODE REVIEW

File: member/models.py [hunk 1]
```diff
@@ -247,4 +253,4 @@ class EmailConfirmation(models.Model):
     language = models.CharField(choices=LANGUAGES, max_length=200, default='en-US')
 
     def as_email_recipient(self):
-        return EmailRecipient(email=self.member.email, language=self.language)
+        return EmailRecipient(email=self.new_email, language=self.language)
```
Comment (bklang) [inline member/models.py:256]: Does this also fix #347 ?

Comment (aaronoppenheimer) [reply member/models.py:256]: It does!

**Comment by aaronoppenheimer** [comment]
The "add a backend to the validation system" was a new idea for me - pretty cool.

**Review by bklang — APPROVED** [review]
LGTM! Tested good locally

**Pull request merged**